nipy/algorithms/clustering/gmm.py

```diff
@@ -217,8 +218,8 @@ def plot2D(x, my_gmm, z=None, with_dots=True, log_scale=False, mpaxes=None,
         if z is None:
             plt.plot(x[:, 0], x[:, 1], 'o')
         else:
-            hsv = plt.cm.hsv(range(256))
-            col = hsv[range(0, 256, 256 // int(z.max() + 1))]
+            hsv = plt.cm.hsv(list(range(256)))
+            col = hsv[list(range(0, 256, 256 // int(z.max() + 1)))]
```

Comment:
Given that range iterables are indexed and reusable, I would recommend (in an off-topic way) dropping that bias, especially if the option is to call `list(range(...))` instead of `range(...)`. `int(x) in range(y)` [is now](https://docs.python.org/dev/library/stdtypes.html#typesseq-range) a O(1) lookup, as opposed to O(n) for `int(x) in list(range(y))`.

More relevant to the current PR, I think I would recommend against indexing numpy arrays by `range` and not `slice`, simply because a `range` is isomorphic to a `slice` (both are jumped-up `(start, stop, step)` tuples), and because there's a built-in slice notation. But perhaps this should be a new PR, since it's more style than notation?